File: Easy/LC231/src/LC231.java
/**
 * Leetoce #231 - https://leetcode.com/problems/power-of-two/
 * 
 * @author ketanmalik June 8 2020
 */

public class LC231 {

	public static void main(String[] args) {
		System.out.println(isPowerOfTwo(512));
	}

	public static boolean isPowerOfTwo(int n) {
		if (n == 1)
			return true;
		if (n <= 0 || (n % 2 != 0))
			return false;
		while (n > 1) {
			if (n % 2 != 0)
				return false;
			n /= 2;
		}
		return true;
	}

}

File: JS/Easy/LC231.js
/**
 * Leetoce #231 - https://leetcode.com/problems/power-of-two/
 *
 * @author ketanmalik June 8 2020
 */

var isPowerOfTwo = function (n) {
  let i = 1;
  while (i < n) i *= 2;
  return i === n;

  // return n.toString(2).match("^10*$") != null;

  // if(n == 1) return true;
  // if(n <= 0 || (n % 2 !== 0)) return false;
  // while(n > 1){
  //     if(n % 2 !== 0) return false;
  //     n /= 2;
  // }
  // return true;
};

console.log(isPowerOfTwo(512));
